src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -533,7 +561,27 @@ private BoundExpression CreateMethodGroupConversion(SyntaxNode syntax, BoundExpr
                 hasErrors = true;
             }
 
-            return new BoundConversion(syntax, group, conversion, @checked: false, explicitCastInCode: isCast, conversionGroup, constantValueOpt: ConstantValue.NotAvailable, type: destination, hasErrors: hasErrors) { WasCompilerGenerated = source.WasCompilerGenerated };
+            if (destination.SpecialType == SpecialType.System_Delegate)
+            {
+                CheckFeatureAvailability(syntax, MessageID.IDS_FeatureInferredDelegateType, diagnostics);
+                // PROTOTYPE: We're resolving the method group multiple times in the code path for a single conversion.
+                var delegateType = GetMethodGroupDelegateType(group, BindingDiagnosticBag.Discarded); // PROTOTYPE: Report use-site diagnostics.
+                var expr = createMethodGroupConversion(syntax, group, conversion, isCast, conversionGroup, delegateType!, hasErrors);
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+                conversion = Conversions.ClassifyConversionFromExpression(expr, destination, ref useSiteInfo);
+                diagnostics.Add(syntax, useSiteInfo); // PROTOTYPE: Report use-site diagnostics.
+                return CreateConversion(syntax, expr, conversion, isCast, conversionGroup, destination, diagnostics);
+            }
+
+#if DEBUG
+            _ = GetMethodGroupDelegateType(group, BindingDiagnosticBag.Discarded);
```

Comment:
Same question here about what this accomplishes. Can't reason out why we would do this in debug only.

Reply:
This was to test inferring the delegate type for all calls to `CreateMethodGroupConversion()`.